app/pages/profile.tsx

```diff
@@ -0,0 +1,228 @@
+
+      <input
+        type="checkbox"
+        id="usernameChangeModal"
+        className="modal-toggle"
+      />
+      <div className="modal">
+        <div className="modal-box">
+          <h3 className="font-bold text-lg">Enter your new display name!</h3>
+          <p className="text-xs">(This name will be visible to other users!)</p>
+          <br />
+          <form action="/profile" onSubmit={handleUsernameChange}>
+            <input
+              type="text"
+              id="newUsername"
+              placeholder={
+                userData?.userName || "Enter your display name here!"
+              }
+              className="input input-bordered w-full max-w-xs"
+            />
+            <div className="modal-action">
+              <label
+                htmlFor="usernameChangeModal"
+                className="btn btn-sm btn-circle absolute right-2 top-2"
+              >
+                ✕
+              </label>
+              <label htmlFor="usernameChangeModal">
+                <button type="submit" className="btn">
+                  Change
+                </button>
+              </label>
+            </div>
+          </form>
+        </div>
+      </div>
+
+      <input type="checkbox" id="imageUploadModal" className="modal-toggle" />
+      <div className="modal">
+        <div className="modal-box relative">
+          <label
+            htmlFor="imageUploadModal"
+            className="btn btn-sm btn-circle absolute right-2 top-2"
+          >
+            ✕
+          </label>
+          <div className="flex flex-col">
+            <h3 className="text-lg font-bold">Upload your Profile Picture!</h3>
+            <div className=" border-4 rounded-lg my-5 w-50 h-50 place-self-center justify-self-center">
+              {imageUpload && (
+                <Image
+                  src={URL.createObjectURL(imageUpload)}
+                  alt="No image"
+                  height={50}
+                  width={50}
+                ></Image>
+              )}
+            </div>
+            {imageError && (
+              <div className="alert alert-error shadow-lg">
+                <div>
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    className="stroke-current flex-shrink-0 w-6 h-6"
+                  >
+                    <path
+                      strokeLinecap="round"
+                      strokeLinejoin="round"
+                      strokeWidth="2"
+                      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
+                    ></path>
+                  </svg>
+                  <span>
+                    File size is too big! &#40;Ensure it is less than 1MB&#41;
+                  </span>
+                </div>
+              </div>
+            )}
+            <input
+              type="file"
+              onChange={(event) => {
```

Comment:
**Nitpick (non-blocking)**: we could move this to a `handleImageUpload` function